app/src/main/java/io/homeassistant/companion/android/settings/SettingsFragment.kt

```diff
@@ -248,15 +249,22 @@ class SettingsFragment : PreferenceFragmentCompat(), SettingsView {
     override fun onDisplayPreferenceDialog(preference: Preference) {
         if (preference is SsidPreference) {
             lateinit var permissionsToCheck: Array<String>
-            if (DisabledLocationHandler.isLocationEnabled(requireContext())) {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+                permissionsToCheck = arrayOf(Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_BACKGROUND_LOCATION)
+            } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                permissionsToCheck = arrayOf(Manifest.permission.ACCESS_COARSE_LOCATION)
+            }
 
-                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
-                    permissionsToCheck = arrayOf(Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_BACKGROUND_LOCATION)
-                } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-                    permissionsToCheck = arrayOf(Manifest.permission.ACCESS_COARSE_LOCATION, Manifest.permission.ACCESS_BACKGROUND_LOCATION)
-                }
+            val fineLocation = DisabledLocationHandler.containsFineLocationPermission(permissionsToCheck)
 
-                if (permissionsToCheck.isNullOrEmpty() || checkPermission(permissionsToCheck, LOCATION_REQUEST_CODE)) {
+            if (DisabledLocationHandler.isLocationEnabled(requireContext(), fineLocation)) {
+                var permissionsToRequest: Array<String>? = null
+                if (!permissionsToCheck.isNullOrEmpty() && Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+                    // For Android 11 we need to remove the Background Location Permission Check
+                    // as for Android 11 the Background Location Request needs to be done separately
```

Comment:
I don't think this should be limited to android 11 in the condition, per androids docs this enforcement is for apps that *target* android 11 so I think it enforces it on everyone right? 

https://developer.android.com/about/versions/11/privacy/location#request-background-location-separately

`If your app targets Android 11 or higher, the system enforces this best practice.`

This sounds like we have to keep it this way for all users right?  Unless I misunderstood this part?

Reply:
Ups this comment confusing. The background permission check is not removed. Only the request for background permission is removed. The code should be fine. Only the comment is wrong. I will fix that comment.

Reply:
So beforehand we never added this check specifically for android 11: https://github.com/home-assistant/android/blob/master/app/src/main/java/io/homeassistant/companion/android/sensors/SensorsSettingsFragment.kt#L122

I wonder if we should just use the best practice on all versions that require background access instead of an additional condition check?

Reply:
Personally i wouldn't do that for all versions, because that would be again another popup for the user. I think on Android 11 these two popups (which are required) are also confusing. But that's just the way it is.

Btw shouldn't this line you mentioned be only for Android >=R instead of >= Q?

Reply:
So background permissions is required for android Q so we do the split there.  To the end user nothing changes.  Foreground and background permissions have always been 2 separate pop-ups.  Now one needs to be granted before the other and we handle that in the code so to the user the only change is the order of granting permissions.  The additional pop-up you see on the onboarding screen is new and requested by Google as a prominent disclosure, this is part of the new policy changes they had.